### PR TITLE
Update field generation and serialization

### LINKED ISSUE #23: Input fields are not created for certain data types
When the contract's abi consists of some 'not expected' types, no input fields are generated. For example, if you have a contract constructor ABI like this:
```json
{
    "type": "constructor",
    "name": "constructor",
    "inputs": [
      {
        "name": "decimals",
        "type": "core::integer::u8"
      },
      {
        "name": "initial_supply",
        "type": "core::integer::u256"
      },
      {
        "name": "recipient",
        "type": "core::starknet::contract_address::ContractAddress"
      }
    ]
},
```
We will have the following output(I have used a storybook to demonstrate):
image
 As you see field `total_supply` is mentioned on the top, but there isn't an input field for it.
If we will change `initial_supply` type from `core::integer::u256` to `core::integer::u128` it will work correctly:
image

So I think the problem is in some kind of type verification.

### PR DESCRIPTION
**Pull request by prxgr4mm3r**
This PR resolves #23 and #25

### CODE REVIEW

File: src/types/helper.ts [hunk 1]
```diff
@@ -159,53 +175,43 @@ export const transformStringArrayToInteger = (value: string[]): bigint[] =>
...
+      return finalizeValues(v);
     });
   }
 
-  if (typeof value === 'object') {
-    return Object.keys(value).reduce((prev, key) => {
-      const curr = value[key];
+  if (typeof val === 'object') {
+    if (Object.keys(val).includes('$type')) {
+      const enumVariant = Object.keys(Object.values(val)[2] as object)[0];
```
Comment (prix0007) [inline src/types/helper.ts]: wrap this in try catch in case it fails

File: src/ABIForm.tsx [hunk 2]
```diff
@@ -4,10 +4,11 @@ import React, { useMemo, useState } from 'react';
 
 import { ABI, abiSchema } from './types/index';
 import {
+  extractEnumsFromABI,
   extractFunctionFromRawAbi,
   extractStructFromABI,
-  segregateViewAndExternalFunctions,
-} from './types/helper';
+  segregateViewAndExternalFunctions
+} from "./types/helper";
```
Comment (satyambnsal) [inline src/ABIForm.tsx]: @prxgr4mm3r I guess your prettier extension is not setup properly. I see it's changing single quotes to double quotes and removing trailing comma as well. Could you please fix it!

File: src/types/uiHelpers.ts [hunk 3]
```diff
@@ -119,6 +154,28 @@ const expandStructsAndReduce = (
...
+
+        if (enumArrIdx > -1) {
+          const reducedEnum = expandEnumsAndReduce(
+            enums[enumArrIdx],
+            enums,
+            structs
+          );
+
+          return {
+            ...pMember,
```
Comment (satyambnsal) [inline src/types/uiHelpers.ts]: Maybe we use descriptive names here instead of `cMember` and `pMember`

**Review by prix0007 — APPROVED** [review]
GTG

**Comment by varex83** [comment]
Can you also increment the version in `package.json`, please?

**Comment by prxgr4mm3r** [comment]
Thank you all for the review! All changes you have requested are done

**Review by satyambnsal — APPROVED** [review]

**Review by varex83 — APPROVED** [review]
LGTM

**Pull request merged**